src/makeCalculator.js

```diff
@@ -37,7 +37,55 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  function add(num) {
+    result += num;
+
+    return this;
+  }
+
+  function subtract(num) {
+    result -= num;
+
+    return this;
+  }
+
+  function multiply(num) {
+    result *= num;
+
+    return this;
+  }
+
+  function divide(num) {
+    result /= num;
+
+    return this;
+  }
+
+  function reset() {
+    result = 0;
+
+    return this;
+  }
+
+  function operate(callback, num) {
+    callback(num);
```

Comment:
Use .call() or .apply() to explicitly set the context of 'this' for the callback function.